plugins/com.google.cloud.tools.eclipse.appengine.libraries.test/src/com/google/cloud/tools/eclipse/appengine/libraries/POMFormatTest.java

```diff
@@ -0,0 +1,134 @@
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.FileLocator;
+import org.eclipse.core.runtime.Platform;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.osgi.framework.Bundle;
+import org.xml.sax.SAXException;
+
+public class POMFormatTest {
+
+  @Rule
+  public final TestProjectCreator projectCreator = new TestProjectCreator();
+  private Pom pom;
+  private IFile pomFile;
+
+  @Before
+  public void setUp() throws SAXException, IOException, CoreException {
+    IProject project = projectCreator.getProject();
+    pomFile = project.getFile("pom.xml");
+    try (InputStream in =
+        Files.newInputStream(Paths.get("testdata/testPomEmpty.xml").toAbsolutePath())) {
+      pomFile.create(in, IFile.FORCE, null);
+      pom = Pom.parse(pomFile);
+    }
+
+    Logger logger = Logger.getLogger(ArtifactRetriever.class.getName());
+    logger.setLevel(Level.OFF);
+  }
+
+
+  @Test
+  public void testAddDependencies() throws Exception {
+    Library library0 = PomTest.newLibrary("id0",
+        new LibraryFile(PomTest.coordinates("com.example.group0", "artifact0", "1.2.3")));
+    Library library1 = PomTest.newLibrary("id1",
+        new LibraryFile(PomTest.coordinates("com.example.group1", "artifact1")));
+    Library library2 = PomTest.newLibrary("id2",
+        new LibraryFile(PomTest.coordinates("com.example.group2", "artifact2")),
+        new LibraryFile(PomTest.coordinates("com.example.group3", "artifact3")));
+
+    List<Library> libraries = Arrays.asList(library0, library1, library2);
+    pom.addDependencies(libraries);
+
+    Bundle bundle = Platform.getBundle("com.google.cloud.tools.eclipse.appengine.libraries.test");
+    URL fileURL = bundle.getEntry("/testdata/formatAdd.xml");
+    File expected = new File(FileLocator.resolve(fileURL).toURI());
+    assertFileContentsEqual(pomFile.getLocation().toFile(), expected);
+
+  }
+
+  @Test
+  public void testRemoveUnusedDependencies() throws Exception {
+    LibraryFile file1 = new LibraryFile(PomTest.coordinates("com.example.group1", "artifact1"));
+    LibraryFile file2 = new LibraryFile(PomTest.coordinates("com.example.group2", "artifact2"));
+    Library library1 = PomTest.newLibrary("id1", file1);
+    Library library2 = PomTest.newLibrary("id2", file1, file2);
+
+    pom.addDependencies(Arrays.asList(library1, library2));
+
+    Bundle bundle = Platform.getBundle("com.google.cloud.tools.eclipse.appengine.libraries.test");
+    URL fileURL = bundle.getEntry("/testdata/formatRemove.xml");
+    File expected = new File(FileLocator.resolve(fileURL).toURI());
+    assertFileContentsEqual(pomFile.getLocation().toFile(), expected);
+  }
+
+  public static void assertFileContentsEqual(File actual, File expected) throws IOException {
+    boolean areEqual = true;
+    try (BufferedReader reader1 = new BufferedReader(new FileReader(actual));
```

Comment:
this depends on system encoding. It needs an InputStreamReader and an explicit encoding, likely UTF-8. However, I think you could make this a lot simpler with Files.readAllLines or Files.readAllBytes.